bundles/core/src/main/java/com/adobe/cq/wcm/core/components/util/ComponentUtils.java

```diff
@@ -70,21 +75,39 @@ public static boolean isDataLayerEnabled(@NotNull final Resource resource) {
             .orElse(false);
     }
 
+
+    /**
+     * See {@link #getId(Resource, Page, String, ComponentContext)}.
+     *
+     * @param resource The resource for which to get or generate an ID.
+     * @param currentPage The current request page.
+     * @param componentContext The current component context.
+     * @return The ID property value for the specified resource, or a generated ID if not set.
+     */
+    @NotNull
+    public static String getId(@NotNull final Resource resource,
+                               @Nullable final Page currentPage,
+                               @Nullable final ComponentContext componentContext) {
+        return getId(resource, currentPage, null, componentContext);
+    }
+
     /**
      * Get the ID property value if set (using {@link #getPropertyId(Resource)},
-     * otherwise generate a new ID (using {@link #generateId(Resource, Page, ComponentContext)}.
+     * otherwise generate a new ID (using {@link #generateId(Resource, Page, String, ComponentContext)}.
      *
      * @param resource The resource for which to get or generate an ID.
      * @param currentPage The current request page.
+     * @param resourceCallerPath The resource caller path(s).
```

Comment:
Is it possible to have multiple paths? Otherwise it should be: The resource caller path.

Reply:
Updated in 4d14728.